legacy/data_spreadsheet.py

```diff
@@ -1,182 +1,167 @@
-        'Biosample_link': 'Provide the Biosample link if available.',
-        'Description': 'Provide an informative description for the Biological_Replicate_ids.',
+from lib.excel import export_to_xlsx
+
+TIME_UNITS = {
+    'd': 'days',
+    'h': 'hours',
+    'm': 'minutes',
+    's': 'seconds',
+}
+
+RED   = 'D02631'
+WHITE = 'FFFFFF'
+
+TECHNIQUE_DESCRIPTIONS = {
+    'fc':          "Flow-cytometry values per time-point in {units}, use a period (.) as the decimal separator",
+    'fc_ps':       "FC counts values per time-point for strain {strain} in {units}, use a period (.) as the decimal separator",
+    'od':          "Optical density values per time-point, use a period (.) as the decimal separator",
+    'plates':      "Plate count values per time-point, use a period (.) as the decimal separator",
+    'plates_ps':   "Plate count values per time-point for strain {strain}, use a period (.) as the decimal separator.",
+    '16s_ps':      "Abundances values per time-point for strain {strain}, use a period (.) as the decimal separator.",
+    'ph':          "pH values per time-point, use a period (.) as the decimal separator",
+    'metabolites': "Concentration values per time-point for metabolite {metabolite} in {units}, use a period (.) as the decimal separator",
+    'STD':         "Standard deviation if more than 1 technical replicate, use a period (.) as the decimal separator",
+}
+
+
+def create_excel(submission, metabolite_names, strain_names):
+    workbook = Workbook()
+
+    metadata_sheet = workbook.active
+    workbook.active.title = "Replicate metadata"
+
+    headers_info = {
+        'Biological_Replicate_id': 'Pick unique identifiers of individual samples: a bottle, a well in a well-plate or mini-bioreactor',
+        'Biosample_link': 'Provide the biosample link if available.',
+        'Description': 'Provide an informative description for the biological replicate',
     }
 
-    for col_idx, (header, description) in enumerate(headers3_constant.items(), start=1):
-        cell = ws3.cell(row=1, column=col_idx, value=header)
-        cell.comment = Comment(description, author="ExcelBot")
-        cell.font = Font(bold=True)  # Make the header bold
-        cell.fill = PatternFill(start_color='FFFFFF', end_color='FFFFFF', fill_type="solid")
-
-        # Calculate column width based on the length of the header text
-        column_width = len(header)
-        column_letter = get_column_letter(col_idx)
-        ws3.column_dimensions[column_letter].width = column_width + 3 # Gives extra space
-
-    # Select the active worksheet
-    wb.create_sheet(title="Growth_Data_and_Metabolites")
-    ws = wb["Growth_Data_and_Metabolites"]
+    for index, (title, description) in enumerate(headers_info.items(), start=1):
+        if index == 1:
+            _add_header(metadata_sheet, index, title, description, fill_color=RED)
```

Comment:
Only the first column (biorep id) is required, so only that one is red. It's difficult to say how important this is, since most columns are going to be required, so they just end up all being red. Maybe a more subtle color might be better.